core/thread.py

```
@@ -113,10 +121,19 @@ def ready(self) -> bool:
         return self._ready_event.is_set()
 
     @ready.setter
-    def ready(self, flag: bool):
+    def ready(self, flag: bool) -> None:
+        """Set the ready state and dispatch thread_create when transitioning to ready.
+
+        Some legacy code paths set thread.ready = True/False. This setter preserves that API by
+        updating the internal event and emitting the creation event when entering the ready state.
+        """
         if flag:
-            self._ready_event.set()
-            self.bot.dispatch("thread_create", self)
+            if not self._ready_event.is_set():
+                self._ready_event.set()
+                try:
+                    self.bot.dispatch("thread_create", self)
+                except Exception:
+                    pass
```

Don't ignore any errors in dispatching. For debugging at least log them.